bonnie-businessrules/src/main/java/com/cgi/hexagon/businessrules/order/OrderService.java

```diff
@@ -75,29 +80,72 @@ public boolean setTrackingNumber(long id, String trackingNr) {
                     order.setStatus(Status.SHIPPED);
                     order.setTrackingNr(trackingNr);
                     order.setLastUpdate(LocalDateTime.now());
-                    if(orderServiceIf.save(order)) {
+                    if (orderStorage.save(order)) {
                         messageService.send(createSendRequest(order));
                         return true;
                     } else {
                         return false;
                     }
                 }
-            } catch(Exception e) {}
+            } catch (Exception e) {
+            }
         }
         return false;
     }
 
     public long createOrder(String productId, int quantity, long assignedTo, Status status) {
-        return orderServiceIf.create(productId, quantity, assignedTo, status);
+        return orderStorage.create(productId, quantity, assignedTo, status);
+    }
+
+    public void createOrders(List<Order> orders) {
+        for (Order order : orders) {
+            createOrder(order);
+        }
+    }
+
+    public long createOrder(Order order) {
+        log.debug("orders arrived to Bonnie :-) ");
+        if (!isNewOrder(order)) {
+            log.error(" False or duplicated orderID in : {}", order.toString());
+            return -1;
+        }
+        if (order.getQuantity() < 1) {
+            log.error(" Invalid quantity in {}", order.toString());
+            return -1;
```

Comment:
If we return with error codes, maybe it is not a bad idea the have distinguishable return codes (-2, -3...)